services/LinkLoss/source/LinkLossService.cpp

```diff
@@ -106,6 +106,7 @@ void LinkLossService::onDataWritten(GattWriteAuthCallbackParams *write_request)
 
     if (level <= (uint8_t)(AlertLevel::HIGH_ALERT)) {
         set_alert_level((AlertLevel) level);
+        write_request->authorizationReply = GattAuthCallbackReply_t::AUTH_CALLBACK_REPLY_SUCCESS;
```

Comment:
that is already set, why set it again?

Reply:
If you look at the simulate_data_written_event() function inside TestLinkLossServiceEvents, the authorisation reply field of the write_request struct is set  AUTH_CALLBACK_REPLY_ATTERR_UNLIKELY_ERROR. Unless the LinkLossService sets the authorisation reply here, assertions about the data written succeeding will fail...

This seems strange to me now that I write it out. Maybe the simulate_data_written_event() function should accept the authorisation reply as a parameter?

Reply:
That seems reasonable @noonfom .